src/datumaro/plugins/data_formats/kinetics.py

```diff
@@ -65,11 +65,15 @@ def _media_files(self, subset):
             return self._subset_media_files[subset]
 
         subset_path = self._subset_path(subset)
-        self._subset_media_files[subset] = {
-            osp.splitext(osp.basename(f))[0]: osp.join(subset_path, f)
-            for f in os.listdir(subset_path)
-            if osp.splitext(osp.basename(f))[1] in VIDEO_EXTENSIONS
-        }
+
+        self._subset_media_files[subset] = {}
+        for root, _, files in os.walk(subset_path):
+            for f in files:
+                file_path = osp.join(root, f)
+                key, file_extension = osp.splitext(osp.basename(file_path))
```

Comment:
Is it just `osp.splitext(f)`?

Reply:
Is it okay to use the base name(without extension) as key? I think it should be a relative path from `subset_path`, because there can be duplication.

Reply:
fixed at ebbdd433a23f4eb474ccddd9ef38540ab6c34f43.

Reply:
You used os.walk(), not os.listdir(). 
If there are files with same basename but different root folder, the value corresponding to the key will be overwritten.
I think this case is technically possible.
So, key should be relative path from subset_path, or some other distinguishable one.

Reply:
Sorry that I failed to understand the point. 
Kinetics directory structure is given by
```
subset
    category
        video_name_start_end.mp4
```
With this structure, I explore all files within a subset folder through `os.walk` and this brings us to `video_name_start_end` as a `key`. Are you worried the situation that there are duplicated `video_name_start_end` files between subsets, e.g., `train` and `val`?

Reply:
You also use subset as key, I think that is distinguishable. 
Let's illustrate an example of a problematic scenario: when bosth 'category/video.mp4' and 'category/some_folder/video.mp4' exist together, or when 'category/video.mp4' and 'category/video.mkv' differ only in their file extensions.
If you're certain that such scenarios don't exist, there's no need to fix this.